src/main/java/seedu/address/model/recurfrequency/RecurFrequency.java

```diff
@@ -1,54 +1,79 @@
 package seedu.address.model.recurfrequency;
 
+import static java.util.Objects.requireNonNull;
+
 import java.util.Arrays;
-import java.util.HashMap;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 import seedu.address.logic.parser.exceptions.ParseException;
 
 public enum RecurFrequency {
-    WEEKLY("W"), BIWEEKLY("BW"), DAILY("D"), MONTHLY("M");
-
-    public static final String DAILY_LABEL = "(Daily)";
-    public static final String WEEKLY_LABEL = "(Weekly)";
-    public static final String BIWEEKLY_LABEL = "(Biweekly)";
-    public static final String MONTHLY_LABEL = "(Monthly)";
-    public static final String INVALID_LABEL = "(Invalid)";
-    private static HashMap<String, RecurFrequency> frequencyMap = new HashMap<>();
+    WEEKLY("W", "WEEKLY"), BIWEEKLY("BW", "BIWEEKLY"),
+    DAILY("D", "DAILY"), MONTHLY("M", "MONTHLY"),
+    NONE("NONE", "NONE");
 
-    static {
-        Arrays.stream(RecurFrequency.values()).forEach(option -> frequencyMap.put(option.frequency, option));
-    }
+    public static final Set<RecurFrequency> VALID_FREQUENCIES =
+            Arrays.stream(RecurFrequency.values()).collect(Collectors.toSet());
     public static final String INVALID_RECUR_FREQUENCY_MESSAGE = String.format("Recurrence "
```

Comment:
Breaking sentence here is awkward. I suggest doing it like this instead
```suggestion
    public static final String INVALID_RECUR_FREQUENCY_MESSAGE = 
            String.format("Recurrence frequency must be one of the following: %s", VALID_FREQUENCIES);"
```